src/pushsource/_impl/model/vms.py

```diff
@@ -57,6 +57,11 @@ class VMIPushItem(PushItem):
     description = attr.ib(type=str, default=None, validator=instance_of_str)
     """A brief human-readable description of the image."""
 
+    boot_mode = attr.ib(
+        type=str, default=None, validator=optional(in_(["hybrid", "uefi", "legacy"]))
+    )
+    """hybrid, uefi, legacy or missing value which means cloud provider's default"""
```

Comment:
bq. Also, I don't think it's appropriate to say "missing value means cloud provider's default". 

If you do not set `BootMode` as part of registering image it will really use AWS provider defaults.
as it is documented in AWS
```
When an AMI has no boot mode parameter, the instances launched from the AMI use the default value of the instance type, which is uefi on Graviton, and legacy-bios on Intel and AMD instance types.
```
* https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ami-boot-mode.html

The feature to cloudimg was introduced just recently
* https://github.com/release-engineering/cloudimg/commit/1ae154580abbeb5ed555d443a2c9bcd61ed1ef70
Without that AWS automatically used `uefi` for aarch64 even though BootMode was not specified at all.
And we shipped aarch64 image in project stratosphere.

But you cannot rely just on AWS defaults. Because it also depends how image was built.
Image builder provides metadata in brew but others which we plan to support later e.g. ART
they needn't provide such information or it can take some to to provide it.
So we still need to have a way how to rely on AWS defaults and thus not set "BootMode" at all as part of registration

Reply:
I've changed it to the suggestion made by @rohanpm